src/main/java/org/javarosa/xpath/XPathConditional.java

```diff
@@ -51,6 +51,11 @@ public Object evalRaw(DataInstance model, EvaluationContext evalContext) {
             if (xpath != null) {
                 throw new XPathUnsupportedException(xpath);
             } else {
+                if (evalContext.isConstraint) {

```

Comment:
@rristovic Any specific reason we need to add this condition ?

Reply:
Hmm on the second thoughts, I there is no specific need. I will remove that check